src/main/java/org/spongepowered/api/event/inventory/InventoryOpenEvent.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.inventory;
+
+import org.spongepowered.api.util.event.Cancellable;
+
+/**
+ * Fired when a viewer opens an inventory.
+ */
+public interface InventoryOpenEvent extends ViewerEvent, Cancellable {
```

Comment:
Players and inventories are kind of weird, especially in the case of Players and their own inventory.  With most inventories, the Player interacts with a Block and then the server sends them the open window packets.  With a Player's own inventory, however, they don't ask for server side validation before opening the Inventory.  The client stores the contents of its own inventory and is therefore able to open the inventory without needing any additional information from the server.  Up until a few minecraft versions ago, the client actually sent nothing to indicate that the Player had opened their inventory.  I'm not sure which version it was that changed it, but the client now sends a packet for the sake of achievements, letting the server know that it opened its inventory.  We experimented around a bit with this in Bukkit, but ultimately it didn't directly get anywhere(the functionality can be achieved via the achievements API)  The best that you can get is closing the Player's inventory right after they open it, which stands in stark contrast to all other inventories where it can simply be prevented.

Also, I should mention that my knowledge base extends only to Minecraft 1.7.  I know nothing about 1.8, and it could be entirely different(Wasn't Dinnerbone rewriting part of inventories?).